packages/gatsby-source-shopify/src/gatsby-node.js

```diff
@@ -25,20 +27,54 @@ import {
 } from "./queries"
 
 export const sourceNodes = async (
-  { actions: { createNode, touchNode }, createNodeId, store, cache },
+  { actions: { createNode, touchNode }, getNodes, createNodeId, store, cache },
   { shopName, accessToken, verbose = true, paginationSize = 250 }
 ) => {
-  const client = createClient(shopName, accessToken)
+  // Arguments used for file node creation.
+  const imageArgs = { createNode, createNodeId, touchNode, store, cache }
 
   // Convenience function to namespace console messages.
   const formatMsg = msg =>
     chalk`\n{blue gatsby-source-shopify/${shopName}} ${msg}`
 
   try {
-    console.log(formatMsg(`starting to fetch data from Shopify`))
+    const online = await isOnline()
+    // if the user sets the GATSBY_SHOPIFY_OFFLINE environment variable
+    // force a fall-back to cached nodes, but never in production
+    const preferCache =
+      process.env.GATSBY_SHOPIFY_OFFLINE === `true` &&
+      process.env.NODE_ENV !== `production`
+
+    if (!online || preferCache) {
+      console.log(chalk`{bold.red Using offline cache for Shopify} ⚠️`)
```

Comment:
It would be great if we would also show timestamp of last fetch.

Reply:
Agreed, this would be really handy. I'm not sure how I'd go about this though. Is there somewhere that you know of where the timestamp of the last fetch is stored, or some easy way for me to add it?

Reply:
We don't really store that information as gatsby doesn't have built in offline capabilities right now. But you can store this yourself using something like
```
cache.set(`last-fetch`, new Date().toJSON())
```

and retrieve it with
```
cache.get(`last-fetch`)
```